interfaces/core.go

```diff
@@ -272,8 +279,9 @@ func StaticInfoOf(iface Interface) (si StaticInfo) {
 	type metaDataProvider interface {
 		StaticInfo() StaticInfo
 	}
-	if iface, ok := iface.(metaDataProvider); ok {
-		si = iface.StaticInfo()
+	if prov, ok := iface.(metaDataProvider); ok {
+		si = prov.StaticInfo()
+		si.defaultDocURL(iface.Name())
```

Comment:
This is the crux of the proposal: If doc URL is not defined by the interface, inject the default URL based on the name.

Reply:
the intention is good but defaultDocURL approach seems a bit clunky. My preference would be do this one level up, in Repository.interfaceInfo. The other approach would be to have a DocumentionURL() method on StaticInfo to use in preference of DocURL. but given that interfaces.Info exists not sure it's better.

Reply:
Went with one level up approach.